<img src="https://full-stack-assets.s3.eu-west-3.amazonaws.com/M03-EDA/Tinder-Symbole.png" width="300">

# Speed Dating with Tinder

## Projet 🚧

Ce projet a pour but de mieux comprendre ce qui fait que les gens sont intéressés les uns par les autres.

L'objectif est de définir les critères qui donnent envie d'avoir un second rendez-vous avec quelqu'un. Pour cela, des rendez-vous ont été organisés et des sondages ont relevé les avis des personnes les unes sur les autres. D'autres questionnaires ont permis d'en savoir plus sur les personnes elles-mêmes.


## Helpers 🦮

To help you achieve this project, here are a few tips that should help you

Data Exploration Ideas :
* What are the least desirable attributes in a male partner? Does this differ for female partners?
* How important do people think attractiveness is in potential mate selection vs. its real impact?
* Are shared interests more important than a shared racial background?
* Can people accurately predict their own perceived value in the dating market?
* In terms of getting a second date, is it better to be someone's first speed date of the night or their last?

# Analyse des données de speed dating

## Plan du notebook 🗺️

1. Découverte et préparation des données
2. Statistiques descriptives
3. Les questions du projet (les cinq questions du sujet, puis "qu'est-ce qui fait dire oui ?")
4. Conclusion

Variables importantes dans tout le notebook :
* dec est la décision de la personne de la ligne. dec = 1 veut dire "oui, je voudrais revoir cette personne".
* dec_o est la décision du partenaire
* match = 1 veut dire que les deux personnes ont dit oui. Il y a donc un match quand dec = 1 et dec_o = 1.

# Partie 1 : découverte et préparation des données

In [1]:
import pandas as pd
import plotly.express as px
import statsmodels

In [2]:
pd.options.display.max_columns = None

data = pd.read_csv("Speed+Dating+Data.csv", encoding="windows-1252")
data.shape

(8378, 195)

In [3]:
data.sample(5, random_state=0)

,iid,id,gender,idg,condtn,wave,round,position,positin1,order,partner,pid,match,int_corr,samerace,age_o,race_o,pf_o_att,pf_o_sin,pf_o_int,pf_o_fun,pf_o_amb,pf_o_sha,dec_o,attr_o,sinc_o,intel_o,fun_o,amb_o,shar_o,like_o,prob_o,met_o,age,field,field_cd,undergra,mn_sat,tuition,race,imprace,imprelig,from,zipcode,income,goal,date,go_out,career,career_c,sports,tvsports,exercise,dining,museums,art,hiking,gaming,clubbing,reading,tv,theater,movies,concerts,music,shopping,yoga,exphappy,expnum,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1,attr4_1,sinc4_1,intel4_1,fun4_1,amb4_1,shar4_1,attr2_1,sinc2_1,intel2_1,fun2_1,amb2_1,shar2_1,attr3_1,sinc3_1,fun3_1,intel3_1,amb3_1,attr5_1,sinc5_1,intel5_1,fun5_1,amb5_1,dec,attr,sinc,intel,fun,amb,shar,like,prob,met,match_es,attr1_s,sinc1_s,intel1_s,fun1_s,amb1_s,shar1_s,attr3_s,sinc3_s,intel3_s,fun3_s,amb3_s,satis_2,length,numdat_2,attr7_2,sinc7_2,intel7_2,fun7_2,amb7_2,shar7_2,attr1_2,sinc1_2,intel1_2,fun1_2,amb1_2,shar1_2,attr4_2,sinc4_2,intel4_2,fun4_2,amb4_2,shar4_2,attr2_2,sinc2_2,intel2_2,fun2_2,amb2_2,shar2_2,attr3_2,sinc3_2,intel3_2,fun3_2,amb3_2,attr5_2,sinc5_2,intel5_2,fun5_2,amb5_2,you_call,them_cal,date_3,numdat_3,num_in_3,attr1_3,sinc1_3,intel1_3,fun1_3,amb1_3,shar1_3,attr7_3,sinc7_3,intel7_3,fun7_3,amb7_3,shar7_3,attr4_3,sinc4_3,intel4_3,fun4_3,amb4_3,shar4_3,attr2_3,sinc2_3,intel2_3,fun2_3,amb2_3,shar2_3,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
2265,165,8.0,1,16,2,7,16,10,12.0,15,2,143.0,0,0.54,1,33.0,2.0,8.33,20.83,20.83,14.58,16.67,18.75,1,10.0,9.0,9.0,10.0,6.0,NaN,9.0,6.0,2.0,30.0,American Studies (Masters),13.0,NaN,NaN,NaN,2.0,4.0,7.0,Detroit,"48,070","52,467.00",3.0,6.0,3.0,Writer,6.0,3.0,1.0,6.0,6.0,7.0,7.0,2.0,1.0,1.0,10.0,8.0,8.0,8.0,5.0,5.0,2.0,2.0,6.0,NaN,16.67,12.50,20.83,18.75,12.50,18.75,10.0,6.0,6.0,8.0,2.0,6.0,9.09,20.45,18.18,20.45,13.64,18.18,8.0,6.0,7.0,9.0,2.0,NaN,NaN,NaN,NaN,NaN,0,6.0,9.0,9.0,6.0,8.0,4.0,6.0,5.0,0.0,3.0,18.18,18.18,22.73,13.64,9.09,18.18,8.0,8.0,9.0,5.0,5.0,8.0,1.0,2.0,NaN,NaN,NaN,NaN,NaN,NaN,19.51,19.51,21.95,14.63,7.32,17.07,9.0,8.0,5.0,7.0,2.0,5.0,13.95,18.6,16.28,16.28,18.6,16.28,8.0,7.0,9.0,5.0,3.0,NaN,NaN,NaN,NaN,NaN,2.0,0.0,1.0,2.0,NaN,17.78,17.78,20.00,15.56,11.11,17.78,NaN,NaN,NaN,NaN,NaN,NaN,9.0,7.0,5.0,7.0,3.0,6.0,5.0,7.0,7.0,8.0,9.0,NaN,8.0,7.0,9.0,5.0,4.0,NaN,NaN,NaN,NaN,NaN
2851,206,13.0,0,25,2,9,20,16,16.0,7,4,217.0,0,0.46,0,26.0,2.0,21.28,17.02,21.28,12.77,6.38,21.28,0,3.0,7.0,7.0,5.0,5.0,4.0,5.0,6.0,2.0,31.0,Public Administration,8.0,NaN,NaN,NaN,1.0,2.0,2.0,Cameroon,NaN,NaN,1.0,5.0,1.0,UN Civil Servant,9.0,5.0,5.0,5.0,9.0,7.0,7.0,7.0,7.0,9.0,9.0,5.0,7.0,7.0,7.0,9.0,5.0,7.0,6.0,NaN,14.58,16.67,20.83,18.75,16.67,12.50,9.0,7.0,8.0,10.0,8.0,9.0,18.00,16.00,18.00,20.00,16.00,12.00,8.0,8.0,8.0,8.0,8.0,NaN,NaN,NaN,NaN,NaN,0,9.0,5.0,9.0,6.0,6.0,5.0,7.0,7.0,2.0,4.0,15.22,19.57,21.74,15.22,13.04,15.22,8.0,9.0,9.0,7.0,8.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,4.0,1.0,1.0,0.0,15.91,15.91,20.45,15.91,15.91,15.91,NaN,NaN,NaN,NaN,NaN,NaN,7.0,7.0,7.0,7.0,7.0,7.0,9.0,7.0,7.0,7.0,7.0,NaN,7.0,9.0,8.0,7.0,9.0,NaN,NaN,NaN,NaN,NaN
3655,256,5.0,0,9,2,11,21,7,7.0,1,2,274.0,0,0.66,0,24.0,3.0,30.00,10.00,20.00,20.00,10.00,10.00,0,4.0,8.0,8.0,5.0,9.0,NaN,6.0,7.0,2.0,24.0,Medicine,4.0,Harvard,"1,400.00","26,019.00",4.0,6.0,5.0,Northern Virginia,"22,015","62,997.00",2.0,6.0,4.0,Physician,4.0,5.0,3.0,6.0,7.0,6.0,5.0,6.0,5.0,5.0,5.0,8.0,6.0,8.0,8.0,8.0,9.0,3.0,6.0,NaN,15.00,25.00,20.00,20.00,10.00,10.00,20.0,15.0,15.0,20.0,10.0,20.0,30.00,10.00,20.00,25.00,5.00,10.00,6.0,9.0,6.0,8.0,6.0,7.0,7.0,8.0,6.0,7.0,0,6.0,7.0,6.0,5.0,6.0,4.0,6.0,7.0,0.0,2.0,20.00,25.00,20.00,10.00,10.00,15.00,7.0,8.0,8.0,7.0,7.0,5.0,1.0,3.0,NaN,NaN,NaN,NaN,NaN,NaN,20.00,20.00,20.00,20.00,10.00,10.00,25.0,15.0,15.0,20.0,15.0,10.0,35.00,10.0,15.00,20.00,5.0,15.00,7.0,8.0,8.0,7.0,7.0,7.0,7.0,8.0,6.0,7.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,

Le tableau contient 8378 lignes et 195 colonnes. Ce que l'on trouve dans ce tableau :
* des informations sur le rendez-vous : la personne, son partenaire, la soirée (wave), le rendez-vous (order), le genre (0 = femme, 1 = homme)
* les décisions : dec (la décision), dec_o (la décision du partenaire), match (les deux ont dit oui)
* les notes données au partenaire juste après le rendez-vous, de 0 à 10 : attr, sinc, intel, fun, amb, shar (Attractive, Sincere, Intelligent, Fun, Ambitious, Shared interests), plus like (j'ai aimé cette personne) et prob (je pense qu'elle a dit oui)
* les mêmes notes mais reçues du partenaire : attr_o, sinc_o, etc.
* les questionnaires, selon le moment où ils ont été remplis : colonnes en _1 (à l'inscription, avant la soirée), en _s (à la moitié de la soirée), en _2 (le lendemain) et en _3 (3 à 4 semaines après avoir)

Décision méthodologique : d'après la Data Key, la question "qu'est-ce que vous cherchez chez l'autre sexe ?" a été posée de deux façons selon la soirée : une note de 1 à 10 pour chaque attribut dans les vagues 6 à 9, et une répartition de 100 points entre les six attributs dans toutes les autres vagues.

Ces deux types de réponses ne sont pas comparables, même quand on ramène chaque total à 100. Avec des notes, les gens ont tendance à donner la même valeur à tous les attributs, alors qu'avec 100 points ils sont obligés de choisir. Les mélanger fausserait les résultats, surtout pour l'attractivité.

On retire donc les vagues 6 à 9 de tout le notebook, pour que toutes les analyses reposent sur les mêmes soirées.

In [4]:
rows_before = len(data)
participants_before = data["iid"].nunique()

df = data[~data["wave"].between(6, 9)].copy()

print("data (before removal of waves 6 to 9):")
print("Number of rows:", rows_before)
print("Number of participants:", participants_before)
print("Number of rows with match:", data["match"].sum())
print("Match ratio:", round(data['match'].sum()/rows_before,3))
print()
print("df (after removal of waves 6 to 9):")
print("Number of rows:", len(df))
print("Number of participants:", df["iid"].nunique())
print("Number of rows with match:", df["match"].sum())
print("Match ratio:", round(df['match'].sum()/len(df),3))


data (before removal of waves 6 to 9):
Number of rows: 8378
Number of participants: 551
Number of rows with match: 1380
Match ratio: 0.165

df (after removal of waves 6 to 9):
Number of rows: 6816
Number of participants: 449
Number of rows with match: 1124
Match ratio: 0.165


La décision méthodologique retire environ 20% des participants. Cela peut sembler beaucoup, mais il reste encore 449 participants et 6816 lignes, ce qui est correct. D'autant plus que le ratio du nombre de match par rapport au total de rendez-vous ne change pas.

On regarde un rendez-vous précis, entre la personne 1 et la personne 11 :

In [5]:
columns_date = ["iid", "pid", "gender", "dec", "dec_o", "match"]

print(df[(df["iid"] == 1) & (df["pid"] == 11)][columns_date])
print(df[(df["iid"] == 11) & (df["pid"] == 1)][columns_date])

   iid   pid  gender  dec  dec_o  match
0    1  11.0       0    1      0      0
     iid  pid  gender  dec  dec_o  match
100   11  1.0       1    0      1      0


On note que le même rendez-vous apparaît sur deux lignes : une ligne vue par la personne 1 et une ligne vue par la personne 11 (dec et dec_o sont inversés, match est identique). Cela fait donc environ 3 408 rendez-vous (6816 lignes divisées par 2), dont 562 avec match (1124 lignes divisées par 2).

In [6]:
important_columns = ["gender", "age", "race", "wave", "order", "pid", "samerace", "dec", "dec_o", "match",
                     "attr", "sinc", "intel", "fun", "amb", "shar", "like", "prob"]

(df[important_columns].isnull().sum())


gender        0
age          90
race         58
wave          0
order         0
pid          10
samerace      0
dec           0
dec_o         0
match         0
attr        194
sinc        253
intel       266
fun         295
amb         584
shar        845
like        212
prob        270
dtype: int64

In [7]:
df.duplicated().sum()

np.int64(0)

Il n'y a aucun doublon. Les décisions (dec, dec_o, match), le genre, la vague et l'ordre sont complets. On remarque quelques valeurs manquantes dans les notes attribuées, l'âge et l'origine ethnique. Le pid (identifiant du partenaire) manque aussi pour 10 lignes.

# Partie 2 : statistiques descriptives

## Le panel : qui sont les participants ?

In [8]:
df["gender_label"] = df["gender"].replace({0: "Women", 1: "Men"}) #pour que les graphiques soient lisibles

persons = df.drop_duplicates("iid") #pour étudier le panel
print("Number of people :", len(persons))

persons["gender_label"].value_counts()

Number of people : 449


gender_label
Men      226
Women    223
Name: count, dtype: int64

In [9]:
persons["age"].describe().round(1)

count    442.0
mean      26.3
std        3.8
min       18.0
25%       23.2
50%       26.0
75%       28.0
max       55.0
Name: age, dtype: float64

In [10]:
persons.groupby("gender_label")["age"].describe().round(1)

,count,mean,std,min,25%,50%,75%,max
gender_label,,,,,,,,
Men,223.0,26.5,3.5,18.0,24.0,27.0,29.0,39.0
Women,219.0,26.1,4.1,19.0,23.0,26.0,28.0,55.0


In [11]:
fig = px.box(persons, x="gender_label", y="age", color="gender_label",
             title="Age of the participants by gender",
             labels={"gender_label": "Gender", "age": "Age"})
fig.update_layout(width=1000)
fig.show()

In [12]:
fig = px.histogram(persons, x="age", color="gender_label", nbins=100, barmode="overlay",
                   title="Age of the participants",
                   labels={"age": "Age", "gender_label": "Gender"})
fig.update_layout(width=1000)
fig.show()

Le groupe est équilibré : 226 hommes et 223 femmes, soit 449 personnes. L'âge moyen (mean) est de 26,3 ans et l'âge médian (median) de 26 ans. La moitié des participants a entre 23 et 28 ans environ (du premier au troisième quartile) et 75 % ont 28 ans ou moins. Les hommes sont très légèrement plus âgés que les femmes (médiane à 27 ans contre 26, soit environ un an d'écart). Les âges des femmes sont plus dispersés (std de 4,1 ans contre 3,5 pour les hommes), en grande partie à cause d'une participante de 55 ans. Le maximum est de 55 ans, bien au-dessus du reste du groupe. L'âge manque pour 7 personnes.

In [13]:
race_names = {1: "Black", 2: "European/Caucasian", 3: "Latino", 4: "Asian", 6: "Other"}

race_counts = persons["race"].replace(race_names).value_counts().reset_index()
race_counts.columns = ["race", "number of participants"]
race_counts

,race,number of participants
0,European/Caucasian,245
1,Asian,112
2,Latino,36
3,Other,32
4,Black,19


In [14]:
fig = px.pie(race_counts, names="race", values="number of participants",
             title="Origin of the participants")

fig.update_traces(textinfo="label+percent")
fig.update_layout(width=800)
fig.show()

Plus de la moitié des participants se déclarent d'origine européenne/caucasienne (245 sur 444 réponses) et environ un quart d'origine asiatique (112). Les autres groupes sont plus petits (Latino 36, autre origine 32, Black 19). Les participants sont liés à l'université Columbia (la Data Key parle de "before coming to Columbia") : l'échantillon ne représente donc pas toute la population.

In [15]:
goal_names = {1: "Fun night out", 2: "Meet new people", 3: "Get a date",
              4: "Serious relationship", 5: "Say I did it", 6: "Other"}

goal_counts = persons["goal"].replace(goal_names).value_counts().reset_index()
goal_counts.columns = ["goal", "number of participants"]
goal_counts

,goal,number of participants
0,Fun night out,180
1,Meet new people,157
2,Get a date,34
3,Say I did it,30
4,Other,26
5,Serious relationship,16


Concernant le but de la soirée (goal), la grande majorité est venue pour s'amuser (180 personnes sur 443 réponses, soit 41 %) ou pour rencontrer de nouvelles personnes (157, soit 35 %). Seulement 34 personnes veulent "avoir un rendez-vous" et 16 cherchent une relation sérieuse. Ce n'est donc pas un public qui cherche à tout prix un couple : on mesure surtout l'envie de revoir quelqu'un après 4 minutes de discussion.

## La cible : à quoi ressemblent les décisions ?

In [16]:
print("Share of rows with dec = 1:", round(df["dec"].mean() * 100, 1), "%")
print("Share of rows with match = 1:", round(df["match"].mean() * 100, 1), "%")

Share of rows with dec = 1: 41.7 %
Share of rows with match = 1: 16.5 %


In [17]:
decision_by_gender = (df.groupby("gender_label")["dec"].mean() * 100).round(1)
decision_by_gender

gender_label
Men      45.5
Women    37.9
Name: dec, dtype: float64

In [18]:
by_gender = (df.groupby("gender_label")[["dec", "match"]].mean() * 100).round(1)
by_gender["yes without match"] = by_gender["dec"] - by_gender["match"]
by_gender = by_gender.rename(columns={"match": "yes with match"})

fig = px.bar(by_gender.reset_index(), x="gender_label", y=["yes with match", "yes without match"],
             title="Share of dates where the participant says yes, by gender (%)",
             labels={"gender_label": "Gender", "value": "Dates (%)", "variable": ""})
fig.update_layout(width=1000)
fig.show()

In [19]:
(df[df["dec"] == 1].groupby("gender_label")["match"].mean() * 100).round(1)

gender_label
Men      36.2
Women    43.6
Name: match, dtype: float64

In [20]:
yes_rate = df.groupby(["iid", "gender_label"])["dec"].mean().reset_index()
yes_rate["dec"].describe().round(2)

count    449.00
mean       0.43
std        0.26
min        0.00
25%        0.24
50%        0.40
75%        0.58
max        1.00
Name: dec, dtype: float64

In [21]:
fig = px.histogram(yes_rate, x="dec", color="gender_label", nbins=80, barmode="overlay",
                   title="Share of yes given by each participant",
                   labels={"dec": "Share of yes given (0 = never, 1 = always)", "gender_label": "Gender"})
fig.update_layout(width=1000)
fig.show()

Au total, 41,7 % des décisions sont des oui, mais seulement 16,5 % des lignes sont des match : il faut que les deux personnes disent oui.

Les hommes disent oui plus souvent que les femmes : 45,5 % contre 37,9 %. La part de oui qui mène à un match est la même pour les deux genres (16,5 % des rendez-vous). La différence vient des oui qui restent sans match : 29,0 % des rendez-vous pour les hommes contre 21,4 % pour les femmes. Parmi ses oui, un homme sur trois environ aboutit à un match (36,2 %), contre 43,6 % pour les femmes. Les femmes sont donc plus sélectives, et ce sont elles qui limitent le nombre de matches.

Le deuxième graphique montre que les participants sont très différents les uns des autres. La part de oui donnée par une personne va de 0 (jamais) à 1 (toujours), avec une moyenne de 0,43 et une écart-type de 0,26. Un quart des participants dit oui à 24 % des personnes ou moins, et un quart à 58 % ou plus. Une partie de ces écarts vient donc probablement du comportement de chacun et pas seulement des qualités des partenaires.

## Les notes par critère : à quoi ressemblent-elles ?

Juste après chaque rendez-vous de 4 minutes, la personne note son partenaire de 0 à 10 sur six critères :
* attr (attractive) : à quel point la personne la trouve attirante
* sinc (sincere) : sincérité
* intel (intelligent) : intelligence
* fun : à quel point elle est amusante, agréable à fréquenter
* amb (ambitious) : ambition
* shar (shared interests) : intérêts communs, les passions et les loisirs que l'on partage

Elle donne aussi deux notes plus générales :
* like : à quel point elle a aimé son partenaire
* prob : la probabilité, selon elle, que son partenaire ait dit oui pour la revoir (ce n'est pas sa propre décision, c'est ce qu'elle croit de la décision de l'autre)


In [22]:
ratings = ["attr", "sinc", "intel", "fun", "amb", "shar"]

df[ratings + ["like", "prob"]].describe().round(2)

,attr,sinc,intel,fun,amb,shar,like,prob
count,6622.00,6563.00,6550.00,6521.00,6232.00,5971.00,6604.00,6546.00
mean,6.16,7.15,7.33,6.38,6.73,5.43,6.12,5.15
std,1.96,1.78,1.56,1.97,1.81,2.18,1.84,2.16
min,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00
25%,5.00,6.00,6.00,5.00,6.00,4.00,5.00,4.00
50%,6.00,7.00,7.00,7.00,7.00,5.00,6.00,5.00
75%,7.00,8.00,8.00,8.00,8.00,7.00,7.00,7.00
max,10.00,10.00,10.00,10.00,10.00,10.00,10.00,10.00


In [23]:
fig = px.box(df, y=ratings,
             title="Ratings given to the partner after each date (0 to 10)",
             labels={"value": "Rating", "variable": "Attribute"})
fig.show()

Toutes les notes vont de 0 à 10. Les attributs les mieux notés en moyenne sont l'intelligence (7,33) et la sincérité (7,15), et ils varient peu d'une personne à l'autre (std de 1,56 et 1,78). À l'inverse, les intérêts communs (moyenne de 5,43), l'attractivité (6,16) et le fun (6,38) ont les moyennes les plus basses et la plus grande dispersion (std de 2,18, 1,96 et 1,97).

Ce sont donc les critères qui différencient le plus les partenaires : presque tout le monde est jugé sincère et intelligent, alors que l'attractivité, le fun et les intérêts communs séparent vraiment les gens.

# Partie 3 : les questions du projet

## Question 1 : quels sont les attributs les moins désirables chez un partenaire masculin ? Est-ce différent chez une partenaire féminine ?

On regarde ce que disent les gens (avant le rendez_vous) et ce qu'ils font (notes données après le rendez_vous).

Avant la soirée, chaque participant a réparti 100 points entre les six attributs, selon ce qu'il cherche chez l'autre sexe (colonnes attr1_1, sinc1_1, etc.). Les réponses des femmes décrivent donc ce qu'elles cherchent chez un homme, et celles des hommes ce qu'ils cherchent chez une femme.

Comme on a retiré les vagues 6 à 9 au début du notebook, toutes les réponses restantes sont des répartitions de 100 points. Vérifions quand même les totaux.

In [24]:
preferences = ["attr1_1", "sinc1_1", "intel1_1", "fun1_1", "amb1_1", "shar1_1"]

prefs = persons[["gender_label"] + preferences].dropna().copy()
total_points = prefs[preferences].sum(axis=1)

total_points.describe().round(2)

count    440.00
mean     100.12
std        2.87
min       90.00
25%      100.00
50%      100.00
75%      100.00
max      148.00
dtype: float64

Pour quelques personnes le total n'est pas exactement 100 (de 90 à 148). On ramène donc chaque total à 100 en divisant par le total de la personne, pour que tout le monde soit comparable.

In [25]:
for column in preferences:
    prefs[column] = prefs[column] / total_points * 100

names = ["Attractive", "Sincere", "Intelligent", "Fun", "Ambitious", "Shared interests"]

stated = prefs.groupby("gender_label")[preferences].mean().T.round(1)
stated.index = names
stated.index.name = "attribute"
stated

gender_label,Men,Women
attribute,,
Attractive,29.1,18.7
Sincere,16.1,18.2
Intelligent,19.4,21.5
Fun,17.4,17.3
Ambitious,7.7,12.0
Shared interests,10.2,12.3


In [26]:
fig = px.bar(stated.reset_index(), x="attribute", y=["Women", "Men"], barmode="group",
             title="What people say they look for in a partner (points out of 100, before the event)",
             labels={"value": "Mean points", "variable": "Who answers"})
fig.update_layout(width=1000)
fig.show()

Ce que disent les 440 personnes qui ont répondu (points sur 100) :
* Pour un partenaire masculin (réponses des femmes), les attributs les moins recherchés sont l'ambition (12,0 points) et les intérêts communs (12,3). Les femmes cherchent surtout l'intelligence (21,5), l'attractivité (18,7) et la sincérité (18,2) ; le fun est à 17,3.
* Pour un partenaire féminin (réponses des hommes), le moins recherché est l'ambition (7,7 points), puis les intérêts communs (10,2). Les hommes donnent un poids très fort à l'attractivité (29,1 points), loin devant l'intelligence (19,4), le fun (17,4) et la sincérité (16,1).

Les deux genres s'accordent donc sur le bas du classement : l'ambition et les intérêts communs sont les attributs les moins recherchés, l'ambition étant encore plus dévalorisée chez une partenaire féminine. La grande différence est l'attractivité : 29,1 points pour les hommes contre 18,7 pour les femmes.

In [ ]:
# calcul de la correlation entre les notes données et la décision, pour voir dans quel mesure l'attribut pèse dans la décision
corr_women = df[df["gender"] == 0][ratings + ["dec"]].corr()["dec"]
corr_men = df[df["gender"] == 1][ratings + ["dec"]].corr()["dec"]

corr_women.index = [names + ['dec']]
corr_men.index = [names + ['dec']]

observed = pd.DataFrame({"Women": corr_women, "Men": corr_men}).round(2)
observed.index.name = "attribute"
observed

,Women,Men
Attractive,0.45,0.50
Sincere,0.22,0.20
Intelligent,0.25,0.21
Fun,0.43,0.40
Ambitious,0.17,0.21
Shared interests,0.43,0.39
dec,1.00,1.00


In [ ]:
fig = px.bar(observed.reset_index(), x="attribute", y=["Women", "Men"], barmode="group",
             title="Correlation between the rating given to the partner and the decision (dec)",
             labels={"value": "Correlation with the decision", "variable": "Who decides"})
fig.show()

Ce que les gens font est un peu différent :
* Pour les deux genres, ce sont l'attractivité (0,45 chez les femmes, 0,50 chez les hommes), le fun (0,43 et 0,40) et les intérêts communs (0,43 et 0,39) qui sont les plus liés à la décision.
* Les attributs les moins liés à la décision sont l'ambition chez les femmes (0,17) et la sincérité chez les hommes (0,20). L'intelligence est aussi faible (0,25 et 0,21), ainsi que l'ambition chez les hommes (0,21).

En résumé, pour un partenaire masculin, l'attribut le moins désirable dans les décisions est l'ambition, puis la sincérité (0,22) et l'intelligence (0,25). Pour un partenaire féminin, c'est la sincérité, puis l'intelligence et l'ambition (0,21 chacune). On remarque que les intérêts communs, parmi les moins cités dans les réponses déclarées, pèsent lourd dans les décisions réelles. C'est le point de départ des deux questions suivantes.

## Question 2 : quelle importance les gens accordent-ils à l'attractivité dans le choix d'un partenaire, par rapport à son impact réel ?

On compare trois mesures de l'importance de chaque attribut, pour chaque genre :
* ce que les gens disent chercher avant la soirée (les points de la question 1)
* l'importance réelle dans leurs décisions. Pour la mesurer, on reprend les corrélations de la question 1 et on les ramène à 100 : le poids d'un attribut est sa corrélation divisée par la somme des six corrélations, fois 100. C'est un indicateur simple, pas une mesure exacte.
* ce que les gens déclarent le lendemain de la soirée, quand on leur demande de répartir 100 points selon ce qui a vraiment compté dans leurs décisions (colonnes attr7_2, sinc7_2, etc.). Peu de personnes ont répondu à cette question.

In [ ]:
weight_women = (corr_women / corr_women.sum() * 100).round(1)
weight_men = (corr_men / corr_men.sum() * 100).round(1)

after = ["attr7_2", "sinc7_2", "intel7_2", "fun7_2", "amb7_2", "shar7_2"]
after_event = persons[["gender_label"] + after].dropna().copy()
total_after = after_event[after].sum(axis=1)

for column in after:
    after_event[column] = after_event[column] / total_after * 100

declared_after = after_event.groupby("gender_label")[after].mean().T.round(1)
declared_after.index = names

print("Participants who answered after the event:")
print(after_event["gender_label"].value_counts())

Participants who answered after the event:
gender_label
Women    62
Men      62
Name: count, dtype: int64


In [ ]:
table_women = pd.DataFrame({"Said before the event": stated["Women"],
                            "Real weight in the decisions": weight_women,
                            "Declared after the event": declared_after["Women"]})
table_women.index.name = "attribute"

table_men = pd.DataFrame({"Said before the event": stated["Men"],
                          "Real weight in the decisions": weight_men,
                          "Declared after the event": declared_after["Men"]})
table_men.index.name = "attribute"

print("Women judging men")
print(table_women)
print()
print("Men judging women")
print(table_men)

Women judging men
                  Said before the event  Real weight in the decisions  \
attribute                                                               
Attractive                         18.7                          23.2   
Sincere                            18.2                          11.5   
Intelligent                        21.5                          12.7   
Fun                                17.3                          21.9   
Ambitious                          12.0                           8.8   
Shared interests                   12.3                          22.0   

                  Declared after the event  
attribute                                   
Attractive                            29.1  
Sincere                               15.2  
Intelligent                           15.8  
Fun                                   17.7  
Ambitious                              8.3  
Shared interests                      13.8  

Men judging women
                  

In [ ]:
columns_compared = ["Said before the event", "Real weight in the decisions", "Declared after the event"]

fig = px.bar(table_women.reset_index(), x="attribute", y=columns_compared, barmode="group",
             title="Women judging men : what they say vs what drives their decisions (out of 100)",
             labels={"value": "Importance (out of 100)", "variable": ""})
fig.show()

fig = px.bar(table_men.reset_index(), x="attribute", y=columns_compared, barmode="group",
             title="Men judging women : what they say vs what drives their decisions (out of 100)",
             labels={"value": "Importance (out of 100)", "variable": ""})
fig.show()

Femmes (qui jugent des hommes) : avant la soirée, elles disent donner 18,7 points sur 100 à l'attractivité. Dans leurs décisions, son poids réel est plus élevé : 23,2 points. Le lendemain, les 62 femmes qui ont répondu à la question en déclarent 29,1. L'attractivité compte donc nettement plus que ce qu'elles pensaient au départ. C'est aussi le cas du fun (17,3 dit, 21,9 réel) et surtout des intérêts communs (12,3 dit, 22,0 réel). À l'inverse, l'intelligence (21,5 dit, 12,7 réel) et la sincérité (18,2 dit, 11,5 réel) pèsent moins que ce qu'elles annonçaient.

Hommes (qui jugent des femmes) : ils disent donner 29,1 points à l'attractivité et son poids réel est un peu plus bas (26,3). Le lendemain, les 62 hommes qui ont répondu en déclarent 35,6. Les hommes semblent donc bien conscients de l'importance de l'attractivité. Comme les femmes, ils sous-estiment les intérêts communs (10,2 dit, 20,3 réel) et surestiment l'intelligence (19,4 dit, 11,2 réel) et la sincérité (16,1 dit, 10,3 réel).

En conclusion, l'attractivité est l'attribut qui pèse le plus dans les décisions des deux genres. Les femmes la sous-estiment dans leurs réponses d'avant la soirée, alors que les hommes l'estiment à peu près correctement. Dans les deux cas, ce que les gens mettent en avant (intelligence, sincérité) pèse moins en pratique que ce qu'ils citent peu (fun, intérêts communs).

Deux précautions. Le poids réel est un indicateur simple, calculé à partir de corrélations, et l'effet de halo (les notes sont fortement liées entre elles, voir la question 6) peut gonfler certains attributs. Et la réponse du lendemain ne vient que de 124 personnes (vagues 15 à 21), elle est donc moins solide.

## Question 3 : les intérêts communs sont-ils plus importants qu'une origine ethnique commune ?

On compare deux colonnes : shar, la note "intérêts communs" donnée après le rendez-vous, et samerace (1 si les deux personnes ont la même origine ethnique, 0 sinon). Pour voir si les deux jouent sur la décision, on regarde la corrélation avec dec et avec match. On ajoute int_corr, qui mesure les intérêts communs de façon objective : c'est la corrélation entre les réponses des deux personnes au questionnaire sur leurs loisirs, remplies avant la soirée.

In [ ]:
df[["dec", "match", "samerace", "shar", "int_corr"]].corr().round(2)

,dec,match,samerace,shar,int_corr
dec,1.00,0.53,0.03,0.41,0.01
match,0.53,1.00,0.03,0.27,0.03
samerace,0.03,0.03,1.00,0.07,0.03
shar,0.41,0.27,0.07,1.00,0.04
int_corr,0.01,0.03,0.03,0.04,1.00


In [ ]:
df["shar_group"] = pd.cut(df["shar"], bins=[-1, 2, 4, 6, 8, 10], labels=["0-2", "3-4", "5-6", "7-8", "9-10"])

rate_by_shar = (df.groupby("shar_group", observed=True)[["dec", "match"]].mean() * 100).round(1)
rate_by_shar["number of dates"] = df.groupby("shar_group", observed=True).size()
rate_by_shar

,dec,match,number of dates
shar_group,,,
0-2,11.6,4.8,688
3-4,20.3,5.8,1128
5-6,42.4,14.9,2194
7-8,65.3,28.7,1552
9-10,76.3,38.1,409


In [ ]:
fig = px.bar(rate_by_shar.reset_index(), x="shar_group", y=["dec", "match"], barmode="group",
             title="Yes decisions and matches (%) depending on the shared interests rating",
             labels={"shar_group": "Shared interests rating (0 to 10)", "value": "Rate (%)", "variable": ""})
fig.show()

In [ ]:
rate_by_race = (df.groupby("samerace")[["dec", "match"]].mean() * 100).round(1)
rate_by_race.index = ["Different ethnic origin", "Same ethnic origin"]
rate_by_race.index.name = "same ethnic origin ?"
rate_by_race

,dec,match
same ethnic origin ?,,
Different ethnic origin,40.7,15.5
Same ethnic origin,43.3,18.0


In [ ]:
fig = px.bar(rate_by_race.reset_index(), x="same ethnic origin ?", y=["dec", "match"], barmode="group",
             title="Yes decisions and matches (%) depending on the ethnic origin",
             labels={"value": "Rate (%)", "variable": ""})
fig.show()

Les participants avaient aussi répondu à la question "à quel point est-il important pour vous que la personne ait la même origine ethnique que vous ?" (colonne imprace, de 0 à 10). Regardons si l'origine commune joue davantage chez ceux qui disent que c'est important.

In [ ]:
df["imprace_group"] = pd.cut(df["imprace"], bins=[-1, 3, 6, 10],
                             labels=["Not important (0-3)", "Medium (4-6)", "Important (7-10)"])

race_effect = (df.groupby(["imprace_group", "samerace"], observed=True)["dec"].mean() * 100).round(1).reset_index()
race_effect["number of dates"] = df.groupby(["imprace_group", "samerace"], observed=True).size().values
race_effect["samerace"] = race_effect["samerace"].replace({0: "Different ethnic origin", 1: "Same ethnic origin"})
race_effect

,imprace_group,samerace,dec,number of dates
0,Not important (0-3),Different ethnic origin,44.8,2551
1,Not important (0-3),Same ethnic origin,44.0,1457
2,Medium (4-6),Different ethnic origin,41.3,782
3,Medium (4-6),Same ethnic origin,45.1,546
4,Important (7-10),Different ethnic origin,27.3,759
5,Important (7-10),Same ethnic origin,40.5,647


In [ ]:
fig = px.bar(race_effect, x="imprace_group", y="dec", color="samerace", barmode="group",
             title="Yes decisions (%) depending on how important the ethnic origin is for the person",
             labels={"imprace_group": "Importance of a shared background (stated by the person)",
                     "dec": "Yes decisions (%)", "samerace": ""})
fig.show()

Les intérêts communs comptent beaucoup plus que l'origine commune.
* La corrélation entre la note shar et la décision est de 0,41 (0,27 avec match), alors que samerace est presque à 0 (0,03 avec dec, 0,03 avec match).
* Le taux de oui passe de 11,6 % pour les rendez-vous notés 0-2 en intérêts communs à 76,3 % pour ceux notés 9-10, et le taux de match de 4,8 % à 38,1 %. À l'inverse, avoir la même origine change peu : 43,3 % de oui contre 40,7 % pour des origines différentes (18,0 % de match contre 15,5 %).

Une nuance : la colonne imprace. Pour les 89 personnes qui disent qu'une origine commune est importante (note de 7 à 10), le taux de oui est de 40,5 % avec une origine commune contre 27,3 % avec une origine différente : leur comportement est cohérent avec leur réponse. Pour les autres, l'écart est faible (de 1 à 4 points, et même en faveur de l'origine différente pour ceux à qui cela importe peu).

Deux précautions. D'abord, shar est une note donnée après le rendez-vous : on note plus haut les intérêts communs d'une personne qu'on apprécie déjà, donc le lien est en partie un effet de halo. Ensuite, int_corr, qui mesure les intérêts communs réels d'après le questionnaire d'avant la soirée, n'est presque pas liée à la décision (0,01). Ce qui compte n'est donc pas d'avoir objectivement les mêmes loisirs, mais de ressentir pendant les 4 minutes qu'on a des choses en commun.

## Question 4 : les gens peuvent-ils prédire avec justesse leur propre valeur sur le marché des rencontres ?

Avant la soirée, chaque personne s'est notée de 0 à 10 sur cinq attributs (colonnes attr3_1, sinc3_1, intel3_1, fun3_1, amb3_1). Pendant la soirée, ses partenaires l'ont notée sur les mêmes attributs (colonnes attr_o, sinc_o, etc.). On compare la note que la personne se donne à la moyenne des notes qu'elle a reçues.

In [ ]:
self_rating = ["attr3_1", "sinc3_1", "intel3_1", "fun3_1", "amb3_1"]
received = ["attr_o", "sinc_o", "intel_o", "fun_o", "amb_o"]
names_5 = ["Attractive", "Sincere", "Intelligent", "Fun", "Ambitious"]

mean_received = df.groupby("iid")[received].mean()
self_vs_received = persons.set_index("iid")[self_rating].join(mean_received)
self_vs_received.head()

,attr3_1,sinc3_1,intel3_1,fun3_1,amb3_1,attr_o,sinc_o,intel_o,fun_o,amb_o
iid,,,,,,,,,,
1,6.0,8.0,8.0,8.0,7.0,6.7,7.4,8.0,7.2,8.000000
2,7.0,5.0,8.0,10.0,3.0,7.7,7.1,7.9,7.5,7.500000
3,8.0,9.0,9.0,8.0,8.0,6.5,7.1,7.3,6.2,7.111111
4,7.0,8.0,7.0,9.0,8.0,7.0,7.1,7.7,7.5,7.700000
5,6.0,3.0,10.0,6.0,8.0,5.3,7.7,7.6,7.2,7.800000


In [ ]:
comparison = pd.DataFrame({"Self rating": self_vs_received[self_rating].mean().values,
                           "Mean rating received": self_vs_received[received].mean().values},
                          index=names_5).round(2)
comparison["gap"] = (comparison["Self rating"] - comparison["Mean rating received"]).round(2)

correlations = []
for own, other in zip(self_rating, received):
    correlations.append(self_vs_received[own].corr(self_vs_received[other]))

comparison["correlation"] = pd.Series(correlations, index=names_5).round(2)
comparison.index.name = "attribute"
comparison

,Self rating,Mean rating received,gap,correlation
attribute,,,,
Attractive,7.01,6.18,0.83,0.29
Sincere,8.25,7.19,1.06,-0.05
Intelligent,8.32,7.37,0.95,0.05
Fun,7.68,6.40,1.28,0.27
Ambitious,7.52,6.78,0.74,0.12


In [ ]:
fig = px.bar(comparison.reset_index(), x="attribute", y=["Self rating", "Mean rating received"],
             barmode="group",
             title="How people rate themselves vs how their partners rate them",
             labels={"value": "Mean rating (0 to 10)", "variable": ""})
fig.show()

In [ ]:
fig = px.scatter(self_vs_received, x="attr3_1", y="attr_o", opacity=0.5, trendline="ols",
                 title="Attractiveness : self rating vs mean rating received (one point per participant)",
                 labels={"attr3_1": "Self rating of attractiveness",
                         "attr_o": "Mean attractiveness rating received"})
fig.show()

Une deuxième mesure : la question expnum, posée seulement lors des premières soirées, demande combien de personnes, parmi celles qu'on va rencontrer, vont avoir envie de nous revoir. On la compare à la part de oui réellement reçus (dec_o).

In [ ]:
yes_received = df.groupby("iid")["dec_o"].mean() * 100

expected = persons.set_index("iid")[["expnum", "round"]].dropna()
expected["expected yes (%)"] = expected["expnum"] / expected["round"] * 100
expected = expected.join(yes_received.rename("received yes (%)"))

print("Number of participants:", len(expected))
print(expected[["expected yes (%)", "received yes (%)"]].mean().round(1))
print("Correlation:", round(expected["expected yes (%)"].corr(expected["received yes (%)"]), 2))

Number of participants: 126
expected yes (%)    47.3
received yes (%)    43.4
dtype: float64
Correlation: 0.25


Les gens se surestiment. Sur les cinq attributs, la note qu'ils se donnent est supérieure d'environ 1 point à la moyenne des notes reçues (+0,74 pour l'ambition, +0,83 pour l'attractivité, +1,28 pour le fun). On ne peut pas savoir si c'est de l'optimisme ou une différence de sévérité entre se noter soi-même et noter les autres, mais l'écart est net.

Au niveau individuel, la corrélation entre l'auto-évaluation et les notes reçues est faible : 0,29 pour l'attractivité, 0,27 pour le fun, 0,12 pour l'ambition, et quasi nulle pour l'intelligence (0,05) et la sincérité (-0,05). Les gens ont donc une vague idée de leur place sur l'attractivité et le fun, mais aucune sur la sincérité et l'intelligence. Ces deux attributs varient peu (tout le monde se met environ 8 et reçoit environ 7), ce qui rend une corrélation difficile à obtenir. Le nuage de points sur l'attractivité le montre : la pente est positive mais faible, avec beaucoup de dispersion.

Sur le nombre de oui attendus (126 personnes, vagues 1 à 5 seulement), la prédiction est bonne en moyenne (47,3 % de oui attendus contre 43,4 % reçus) mais faible individuellement (corrélation de 0,25).

Conclusion : les gens ne prédisent pas bien leur valeur sur le marché de la rencontre. Ils se surestiment et ne se situent à peu près correctement que sur l'attractivité et le fun.

## Question 5 : pour obtenir un second rendez-vous, vaut-il mieux être le premier rendez-vous de la soirée de quelqu'un, ou le dernier ?

La colonne order donne le numéro du rendez-vous dans la soirée. Pour une personne, "obtenir un second rendez-vous" veut dire que son partenaire a dit oui (dec_o = 1), ou qu'il y a match. Les soirées n'ont pas toutes le même nombre de rendez-vous : on repère le dernier numéro de chaque soirée pour pouvoir parler du dernier rendez-vous.

In [ ]:
wave_max = df.groupby("wave")["order"].max()
df["wave_max"] = df["wave"].map(wave_max)

first_date = df[df["order"] == 1]
last_date = df[df["order"] == df["wave_max"]]

comparison_order = pd.DataFrame({"First date": first_date[["dec_o", "match"]].mean() * 100,
                                 "Last date": last_date[["dec_o", "match"]].mean() * 100,
                                 "All dates": df[["dec_o", "match"]].mean() * 100}).round(1)

print("Number of first dates:", len(first_date), " Number of last dates:", len(last_date))
comparison_order

Number of first dates: 449  Number of last dates: 352


,First date,Last date,All dates
dec_o,48.6,44.9,41.7
match,22.9,21.0,16.5


In [ ]:
rate_by_order = (df.groupby("order")[["dec_o", "match"]].mean() * 100).round(1)
rate_by_order["number of dates"] = df.groupby("order").size()
rate_by_order.head(12)

,dec_o,match,number of dates
order,,,
1,48.6,22.9,449
2,39.6,15.6,449
3,40.8,17.1,449
4,43.9,18.0,449
5,43.9,16.7,449
6,43.4,17.4,449
7,41.0,17.3,422
8,42.1,16.6,416
9,45.6,20.2,410


In [ ]:
fig = px.line(rate_by_order.reset_index(), x="order", y=["dec_o", "match"],
              title="Yes received and matches (%) depending on the number of the date in the evening",
              labels={"order": "Number of the date in the evening", "value": "Rate (%)", "variable": ""})
fig.show()

Pour les numéros élevés, il y a peu de rendez-vous (seules les grandes soirées vont jusque là), donc la courbe devient irrégulière. Pour lisser, on regroupe les rendez-vous par quart de soirée, avec la position relative = numéro du rendez-vous divisé par le dernier numéro de la soirée.

In [ ]:
df["relative_position"] = df["order"] / df["wave_max"]
df["quarter"] = pd.cut(df["relative_position"], bins=[0, 0.25, 0.5, 0.75, 1],
                       labels=["1st quarter", "2nd quarter", "3rd quarter", "4th quarter"])

rate_by_quarter = (df.groupby("quarter", observed=True)[["dec_o", "match"]].mean() * 100).round(1)
rate_by_quarter["number of dates"] = df.groupby("quarter", observed=True).size()
rate_by_quarter

,dec_o,match,number of dates
quarter,,,
1st quarter,43.0,18.5,1543
2nd quarter,42.4,17.5,1880
3rd quarter,41.6,14.6,1658
4th quarter,39.7,15.4,1735


In [ ]:
fig = px.bar(rate_by_quarter.reset_index(), x="quarter", y=["dec_o", "match"], barmode="group",
             title="Yes received and matches (%) depending on the moment of the evening",
             labels={"quarter": "Moment of the evening", "value": "Rate (%)", "variable": ""})
fig.show()

Être le premier rendez-vous de la soirée est un léger avantage : 48,6 % des premiers rendez-vous reçoivent un oui (22,9 % de match), contre 41,7 % (16,5 %) pour l'ensemble des rendez-vous. Le dernier rendez-vous n'a pas d'avantage comparable : 44,9 % de oui et 21,0 % de match, soit à peine mieux que la moyenne.

Sur la courbe, le taux est le plus haut au numéro 1, puis il oscille entre 34 % et 46 % environ sans tendance nette (les numéros élevés sont plus irréguliers, car il y a moins de rendez-vous). Par quart de soirée, le taux de oui baisse de 43,0 % (premier quart) à 41,6 % (troisième quart) puis 39,7 % (dernier quart) : l'écart est d'environ 3 points.

Réponse : il vaut mieux être le premier que le dernier, mais l'effet est petit. Avec 449 premiers rendez-vous et 352 derniers, la marge d'incertitude est de l'ordre de 5 points : l'écart de 3,7 points entre le premier et le dernier n'est pas très net, alors que celui avec la moyenne (6,9 points) l'est davantage. L'ordre de passage compte très peu comparé aux intérêts communs : le taux de oui passe de 11,6 % à 76,3 % entre les notes 0-2 et 9-10.

## Question 6 : qu'est-ce qui fait dire oui ?

Première approche : comparer les notes données au partenaire quand la personne dit non (dec = 0) et quand elle dit oui (dec = 1).

In [ ]:
mean_by_decision = df.groupby("dec")[ratings].mean().T.round(2)
mean_by_decision.columns = ["No (dec = 0)", "Yes (dec = 1)"]
mean_by_decision["difference"] = (mean_by_decision["Yes (dec = 1)"] - mean_by_decision["No (dec = 0)"]).round(2)
mean_by_decision

,No (dec = 0),Yes (dec = 1),difference
attr,5.35,7.25,1.90
sinc,6.82,7.58,0.76
intel,7.03,7.74,0.71
fun,5.67,7.32,1.65
amb,6.45,7.12,0.67
shar,4.65,6.46,1.81


In [ ]:
mean_by_decision.index.name = "attribute"

fig = px.bar(mean_by_decision.reset_index(), x="attribute", y=["No (dec = 0)", "Yes (dec = 1)"],
             barmode="group",
             title="Mean rating given to the partner, depending on the decision",
             labels={"value": "Mean rating (0 to 10)", "variable": "Decision"})
fig.show()

Quand la personne dit oui, elle a donné de meilleures notes sur tous les attributs, mais l'écart (difference) n'est pas le même partout. Il est le plus grand pour l'attractivité (+1,90), les intérêts communs (+1,81) et le fun (+1,65). Il est beaucoup plus faible pour la sincérité (+0,76), l'intelligence (+0,71) et l'ambition (+0,67).

Autrement dit, ce qui sépare un oui d'un non, ce sont d'abord l'attractivité, le fun et les intérêts communs. Même les partenaires à qui l'on dit non reçoivent des notes élevées en sincérité (6,82) et en intelligence (7,03) : ces qualités ne départagent pas.

In [ ]:
corr_matrix = df[ratings + ["like", "prob", "dec", "match"]].corr().round(2)

fig = px.imshow(corr_matrix, text_auto=True, color_continuous_scale="RdBu_r", zmin=-1, zmax=1,
                title="Correlation between the ratings and the decisions")
fig.update_layout(width=1000)
fig.show()

On lit ce qui est lié à la décision dans la colonne dec :
* like (0,51), l'attractivité (0,48), le fun (0,42) et les intérêts communs (0,41) ont les corrélations les plus fortes avec le oui. like est une impression générale sur la personne, pas un attribut, donc il est logique qu'elle soit la plus proche de la décision.
* la sincérité (0,21), l'intelligence (0,23) et l'ambition (0,18) sont nettement plus faibles.
* prob (0,32), la probabilité estimée que l'autre dise oui, est aussi liée à la décision : on a plus envie de revoir quelqu'un qui semble nous apprécier.

La corrélation avec match est plus faible (par exemple 0,26 pour l'attractivité), parce qu'un match demande deux oui : la note d'une seule personne n'explique qu'une moitié de l'histoire.

Une précaution : les notes sont fortement liées entre elles (0,67 entre attr et like, 0,58 entre attr et fun, 0,65 entre sinc et intel). Ces corrélations montrent donc des liens, pas des causes.

# Partie 4 : conclusion

Ce que montrent les données du speed dating pour comprendre ce qui pousse à dire oui :
* Le oui dépend d'abord de l'attractivité (corrélation de 0,48 avec la décision), du fun (0,42) et des intérêts communs ressentis (0,41). La sincérité, l'intelligence et l'ambition comptent beaucoup moins (0,18 à 0,23).
* Les hommes disent oui plus souvent que les femmes (45,5 % contre 37,9 %). Seulement 16,5 % des lignes sont des match, car il faut deux oui.
* Ce que les gens disent chercher ne correspond pas à ce qu'ils font. Avant la soirée, les femmes valorisent l'intelligence et la sincérité, alors que ce sont l'attractivité, le fun et les intérêts communs qui pèsent dans leurs décisions. Les hommes sont plus lucides sur l'attractivité mais sous-estiment aussi les intérêts communs.
* Les intérêts communs ressentis comptent beaucoup plus que l'origine commune (corrélation de 0,41 contre 0,03), sauf pour les personnes qui disent que l'origine est importante.
* Les gens se surestiment d'environ 1 point sur 10 et situent mal leur valeur, surtout sur la sincérité et l'intelligence.
* L'ordre de passage joue très peu : un petit avantage pour le premier rendez-vous (49 % de oui contre 42 % en moyenne).

Pistes pour l'équipe marketing de Tinder :
* Les éléments qui donnent une impression d'attractivité et de fun (photos, ton de la bio) sont ceux qui pèsent le plus dans la décision.
* Faire ressortir les centres d'intérêt dans les profils peut aider, car l'impression d'avoir des choses en commun est fortement liée au oui, même si les loisirs communs mesurés objectivement ne le sont pas.
* Les préférences déclarées par les utilisateurs sont un guide fragile : leurs choix réels sont différents. Le comportement (les swipes) semble plus fiable que ce qu'ils disent chercher.

Limites :
* L'échantillon est petit (449 étudiants d'une même université, entre 2002 et 2004) et un rendez-vous de 4 minutes en face à face n'est pas un swipe sur une application.
* Quatre soirées (vagues 6 à 9) ont été écartées, car leurs préférences déclarées ne sont pas comparables à celles des autres soirées.
* Les notes sont données après le rendez-vous et s'influencent entre elles (effet de halo).
* Les corrélations montrent des liens, pas des causes.
* On a utilisé des statistiques descriptives, sans test statistique.